opam-state.opam

```diff
@@ -26,7 +26,7 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 depends: [
-  "ocaml" {>= "4.03.0"}
+  "ocaml" {>= "4.08.0"}
```

Comment:
It is to be noted that notable plugins, such as `dune-release` and `opam-publish`, use `opam-state`.

Not a blocker ofc but it's probably good to ping the devs. cc @NathanReb

Reply:
Indeed - I'd checked those through before proposing it. opam-publish is already pulled towards 4.08+ by depending on cohttp via ocaml-github and dune-release was already 4.06+ (and I think @emillon agreed it would shift on @NathanReb's behalf!)

Reply:
I confirm there's no problem on my side for bumping minimum requirements on OCaml, to the contrary I'm rather glad about it!